### PR TITLE
HTML Validator unit test

### PR DESCRIPTION
**Pull request by siuster95**

### CODE REVIEW

File: test/unit/htmlValidatorTest.js [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+        // check to see tha tthe page did not validate
+        let test1 = res.text.includes('HTML did not pass validation')
+        assert.equal(test1, true)
+        testApp.kill()
+        done()
+      })
+    })
+  })
+
+  it('should not validate the html page when the request header sent to it has the exception value', function (done) {
```
Comment (Autre31415) [inline test/unit/htmlValidatorTest.js]: Could you put this test next to the other header exception tests

File: test/unit/htmlValidatorTest.js [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+      onServerStart: true
+    }, options)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), ['--dev'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // on the message that we get back that the server has started, test the htmlValidator by trying to recieve a bad html page
+    testApp.on('message', () => {
+      // request the bad html page
+      request('http://localhost:43711')
```
Comment (kethinov) [inline test/unit/htmlValidatorTest.js]: don't hardcode the port on these. default could change some day which would require us to update all the tests

File: test/unit/htmlValidatorTest.js [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+      onServerStart: true
+    }, options)
+
+    // fork the app and run it as a child process
+    const testApp = fork(path.join(appDir, 'app.js'), ['--dev'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    // on the message that we get back that the server has started, test the htmlValidator by trying to recieve a bad html page
+    testApp.on('message', (params) => {
+      // request the bad html page
+      console.log(params.port)
```
Comment (kethinov) [inline test/unit/htmlValidatorTest.js]: gotta remove the log

File: test/unit/htmlValidatorTest.js [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+        testApp.kill('SIGINT')
+      })
+
+      // check to see if the validator is accessible (should get back 200)
+      request(`http://localhost:${params.htmlValidator.port}`)
+      .get('/')
+      .expect(200, (err, res) => {
+        if (err) {
+          assert.fail(err)
+          const killLine = fork(path.join(__dirname, '../', '../', 'lib', 'scripts', 'killValidator.js'))
```
Comment (Autre31415) [inline test/unit/htmlValidatorTest.js]: You don't have to worry about path.join when using this fork api. Also if you pass the same params to `killValidator` that you pass to the testApp fork there won't be any logging.

Comment (kethinov) [reply test/unit/htmlValidatorTest.js]: fix the merge conflict too

**Pull request merged**